embassy-stm32/src/lib.rs

```diff
@@ -45,6 +45,12 @@ pub struct Config {
     rcc: rcc::Config,
 }
 
+impl Config {
+    pub fn new(rcc: rcc::Config) -> Self {
```

Comment:
I wonder how this will work with macro's though. Currently, the way to configure clock would be:

```rust
#[embassy::main(config = "embassy_stm32::Config::new(embassy_stm32::rcc::Config::hsi16())")]
```

Having to use `let` and `blah=foo` is not going to work with that, I think. Would it work if using "builders"? I.e.

```rust
#[embassy::main(config = "embassy_stm32::Config::default().rcc(embassy_stm32::rcc::Config::default().clock_src(ClockSrc::HSI16))")]
```


I've pushed a change with this approach.